squid-forward-proxy-operator/src/http_proxy.py

```diff
@@ -0,0 +1,882 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""HTTP proxy charm library."""
+
+import copy
+import ipaddress
+import json
+import re
+import uuid
+from typing import TYPE_CHECKING, Dict, Iterable, List, Optional, Sequence, Tuple, cast
+
+import ops
+from pydantic import BaseModel, SecretStr, conint, field_validator, model_validator
+
+AUTH_METHOD_SRCIP_USERPASS = "srcip+userpass"
+AUTH_METHOD_USERPASS = "userpass"
+AUTH_METHOD_SRCIP = "srcip"
+AUTH_METHOD_NONE = "none"
+AUTH_METHODS = [
+    AUTH_METHOD_SRCIP_USERPASS,
+    AUTH_METHOD_USERPASS,
+    AUTH_METHOD_SRCIP,
+    AUTH_METHOD_NONE,
+]
+PROXY_STATUS_PENDING = "pending"
+PROXY_STATUS_ACCEPTED = "accepted"
+PROXY_STATUS_DENIED = "denied"
+PROXY_STATUS_INVALID = "invalid"
+PROXY_STATUS_ERROR = "error"
+PROXY_STATUS_READY = "ready"
+PROXY_STATUSES = [
+    PROXY_STATUS_PENDING,
+    PROXY_STATUS_ACCEPTED,
+    PROXY_STATUS_DENIED,
+    PROXY_STATUS_INVALID,
+    PROXY_STATUS_ERROR,
+    PROXY_STATUS_READY,
+]
+
+NO_CHANGE = object()
+UnsignedInt = conint(ge=0)
```

Comment:
Use Annotated instead

Reply:
Use `Annotated[int, Field(ge=0)]` instead. Indeed, this is better, thanks @yhaliaw!